src/common/lib/command_mode.c

```
@@ -170,462 +180,664 @@ static uint16_t brightness_rainbow_hue = 0;      /**< Current rainbow hue (0-359
+    return false;
+}
 #endif
-  }
-  
-  // Should never reach here, but default to normal processing
-  return true;
+
+bool command_mode_process(const hid_keyboard_report_t* keyboard_report) {
+    // Dispatch to per-state handler based on current state
+    switch (cmd_mode.state) {
+        case CMD_MODE_IDLE:
```

Clang (14.0.6)

[warning] switch has 5 consecutive identical branches

(bugprone-branch-clone)